pkg/controller/knativeserving/openshift/openshift.go

```diff
@@ -491,3 +497,27 @@ func augmentAutoscalerDeployment(u *unstructured.Unstructured) error {
 	}
 	return nil
 }
+
+// Update logging URL template for Knative service's revision with concrete kibana hostname if cluster logging has been installed
+func kibana(u *unstructured.Unstructured) error {
+	if u.GetKind() == "ConfigMap" && u.GetName() == "config-observability" {
+		// attempt to locate kibana route which is available if openshift-logging has been configured
+		route := &routev1.Route{}
+		if err := api.Get(context.TODO(), types.NamespacedName{Name: "kibana", Namespace: "openshift-logging"}, route); err != nil {
+			if !meta.IsNoMatchError(err) {
```

Comment:
This should actually ignore `IsNotFoundError`. `NoMatchError` will happen if a specific CRD (Route in this case) is not installed, which will never be the case in an Openshift installation.

Reply:
@markusthoemmes  Hmm .... your point actually is an argument for logging an error because an expected pre-condition has not been met.   WDYT?

Reply:
Don't we automatically log the error returned on a failed reconcilation anyway? I'm not a fan of logging deeply inside the callstack.

Reply:
If we ignore IsNotMatchError (and return nil), it will not fail the reconciliation.